server/machines/machines.controller.go

```diff
@@ -0,0 +1,34 @@
+package machines
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type HttpHandlerFunc http.HandlerFunc
+
+func HttpHandler(ms *MachinesService) HttpHandlerFunc {
+	return func(rw http.ResponseWriter, r *http.Request) {
+		if r.Method == "GET" {
+			handleListMachines(rw, ms)
+		} else {
+			rw.WriteHeader(http.StatusMethodNotAllowed)
+		}
+	}
+}
+
+func handleListMachines(rw http.ResponseWriter, ms MachinesService) {
+	rw.Header().Set("content-type", "application/json")
+	res, err := ms.ReadAll()
+	if err != nil {
+		rw.WriteHeader(404)
+		_ := json.NewEncoder(rw).Encode("cannot read all machines")
+		return
```

Comment:
If no entries - return `[]`, not `error`

Reply:
if no entries returns `[]`

Reply:
If you have error then return 500, not 404